addons/mail/static/src/messaging/entity/activity/activity.js

```diff
@@ -13,22 +13,20 @@ function ActivityFactory({ Entity }) {
 
     class Activity extends Entity {
 
-        /**
-         * @override
-         */
-        delete() {
-            this.env.rpc({
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        async deleteRecord() {
```

Comment:
doc

Reply:
> Last detail on current diff below, but we need to add a test for that feature too.

The markdone or the deleteRecord ?

If I remember well, I had to do less test in activity because entity was coming but I'm not sure anymore, I just remember I had to stop testing to do something else so IMO activity could have some more tests.

Reply:
Well now that you ask... Both I suppose. Runbot should not be green when any feature is not working.

Reply:
deleteRecord in implicitely tested in `activity_tests.js`
As we don't have "entity tests", I don't know how you could test `deleteRecord` specifically ?

Reply:
If `deleteRecord` is called by a component, I'm fine if we just test the component. But it is not testing enough since this fix is needed but runbot is green.
I suppose there should be a test mocking the RPC to fail and testing that it is not visually deleted.

Reply:
Note that testing something doesn't happen is currently very annoying (because there is no new render and no hint to tell us we waited "long enough").

Reply:
In fact there is no test for mark as done and schedule next and mark as done test do not test that a rpc call is made. 'gonna cure that